### DISCUSSION TITLE
Serving a static frontend with app.mount()

### DISCUSSION THREAD

**Discussion by pietz** [fastapi/fastapi#5443] [Questions] [answered]
Hey team,

I'm currently building a web app with FastAPI in the backend and Svelte in the frontend. Since I don't require any advanced frontend features, it compiles down to a static website I want to deploy. Most articles on web development clearly separate frontend and backend deployments. This makes complete sense in larger projects to decouple the two.

In my small scale setup, is there anything wrong with deploying the frontend as a fastapi route like this?

```python
app.mount("/", StaticFiles(directory="build", html=True), name="frontend")
```

I figured that I can always separate them later without much hassle and for the time being it's just super nice to only manage one cloud resource. The reason I'm asking is that I haven't found any other resources or tutorials doing something similar.

Is there a reason for it? Am I completely missing something?

**Comment by acidjunk** [comment, marked as answer]
I use it to serve docs : e.g. we have mkdocs in our fastapi project to document some stuff. The output is a folder with html.
Very convenient to don't need an extra webserver for that.

For larger sites a CDN is better as it can cache all the static files, until next deployment, and removes some load from the api backend. Or a very simple extra NginX server that serves the static files.

**Comment by Christianjade1** [comment]
I tried to tell people but they never listen..  Nobody will help me but if
they cared even a little bit this would've been avoided..
